src/api/app/models/cloud/backend/upload_job.rb

```diff
@@ -0,0 +1,34 @@
+module Cloud
+  module Backend
+    class UploadJob
+      include ActiveModel::Validations
+
+      attr_reader :id, :state, :details, :platform, :project_name, :package_name, :repository_name, :architecture_name, :file_name, :backend_response
+      validate :validate_response
+
+      def initialize(xml)
+        xml_hash = Xmlhash.parse(xml)
+        self.id = xml_hash['name']
+        self.state = xml_hash['state']
+        self.details = xml_hash['details']
+        self.platform = xml_hash['target']
+        self.project_name = xml_hash['project']
+        self.package_name = xml_hash['package']
+        self.repository_name = xml_hash['repository']
+        self.architecture_name = xml_hash['arch']
+        self.file_name = xml_hash['filename']
```

Comment:
Some suggestions for this class:
1. Move the code that fetches the xml from the backend to this class, so we have all backend related code/logic in one class.
2. Store only the raw xml as an instance variable and have the rest of the attributes (id, state, details etc.) be derived from the raw xml, so you don't have to override the `initialize` method that ActiveModel already provides.

#2 is a nitpick, I just think its better not to override the `initialize` method ActiveModel gives you and its easier to make the class flexible (i.e. you can make it so that if you update the xml then the attributes get automatically updated too).

As an example of what I'm talking about the class could be like this:
https://gist.github.com/evanrolfe/6216aeac331db5d54dbf2562ac4fb5c4